### ISSUE TITLE
Cycle animators cannot target vector types

### ISSUE THREAD

**Issue by Sunjammer** [fusetools/fuselibs-public#348]
It seems odd from a user pov that `Change foo.Color="White"` works while `Cycle Target="foo.Color" Low="White" ...` does not.  This also makes a pretty common use case for cycle animators unnecessarily awkward, since color cycling for highlight strokes and more is super useful and when you have to split your colors from, say, a palette of globals, into individual components per Cycle animator, the whole process becomes messy, and makes the UX look pretty bad as well.

**Comment by mortoray** [comment]
Cycle needs a linear range to cycle on, thus it's expecting only scalars for high/low. To use on non-scalars you need to use the `Base` and `Offset` values, which convert the scalar into the vector type. Refer to docs on `Cycle`.

We probably could extend the system such that it does this automatically though. If you assigned a vector to both Low/High it could internally compute the Base, Offset, and Low and High from it. I didn't do this before though since the code didn't make it easy and it's just not been a high enough priority.

**Comment by Sunjammer** [comment]
Yeah I was just looking at finding a good way to use Base/Offset and I just don't find it very intuitive. At first I thought the Base would be the value at 0 and the offset would be the value added to Base when at 1, but instead I needed to do the inverse to achieve what I wanted.

`<Cycle Target="stroke.Color" Base="White" Offset="SwitchTeal" Waveform="Triangle" Low="0" High="1" Frequency="0.5"/>
`

In this scenario I wanted to animate between SwitchTeal and White, and intuitively I figured the teal color would be my base.

It would be really nice if we could make it automatic :) It's a really useful feature

**Comment by mortoray** [comment]
You can use UX Expressions now to get the correct offset: `Base="Teal" Offset="White-Teal"`.

The current wacky API comes primarily from limitations in the UX/Uno compiler when `Cycle` was written (I think they've been gone for a while, but backwards compatibility still makes it a tad annoying to change). Nobody ever really thought this Base/Offset idea was a good thing. :)

**Comment by Sunjammer** [comment]
By the way, doing White-Teal for the offset gives me unexpected results, cycling on a much wider range
